python/ray/experimental/workflow/workflow_manager.py

```diff
@@ -169,19 +158,65 @@ def execute(self) -> WorkflowOutputType:
         """
         if self.executed:
             return self._output
-        input_args = []
-        input_kwargs = {}
-        for a in self.args:
-            if isinstance(a, Workflow):
-                input_args.append(a.execute())
-            else:
-                input_args.append(a)
-        for k, v in self.kwargs.items():
-            if isinstance(v, Workflow):
-                input_kwargs[k] = v.execute()
-            else:
-                input_kwargs[k] = v
-        output = self.step_func(input_args, input_kwargs)
+        workflow_outputs = [w.execute() for w in self._captured_workflows]
```

Comment:
Will this disallowing steps from running in parallel?

Reply:
no, `execute()` is non-blocking